src/VisualStudio/Core/Def/Packaging/PackageSearchService.cs

```diff
@@ -0,0 +1,614 @@
+                var result = XElement.Load(stream);
+                LogInfo("Converting data to XElement completed");
+                return result;
+            }
+        }
+
+        private async Task RepeatIOAsync(Action action)
+        {
+            const int repeat = 6;
+            for (var i = 0; i < repeat; i++)
+            {
+                _cancellationToken.ThrowIfCancellationRequested();
+
+                try
+                {
+                    action();
+                    return;
+                }
+                catch (Exception e) when (_swallowException(e))
+                {
+                    var delay = _delayService.FileWriteDelay;
+                    LogException(e, $"Operation failed. Trying again after {delay}");
+                    await Task.Delay(delay, _cancellationToken).ConfigureAwait(false);
+                }
+            }
+        }
+
+        private byte[] ParseDatabaseElement(XElement element)
+        {
+            LogInfo("Parsing database element");
+            var contentsAttribute = element.Attribute(ContentAttributeName);
+            if (contentsAttribute == null)
+            {
+                throw new FormatException($"Database element invalid. Missing '{ContentAttributeName}' attribute");
+            }
+
+            var text = contentsAttribute.Value;
+            var compressedBytes = Convert.FromBase64String(text);
+
+            using (var inStream = new MemoryStream(compressedBytes))
+            using (var deflateStream = new DeflateStream(inStream, CompressionMode.Decompress))
+            using (var outStream = new MemoryStream())
+            {
+                deflateStream.CopyTo(outStream);
+                var bytes = outStream.ToArray();
+
+                LogInfo($"Parsing complete. bytes.length={bytes.Length}");
+                return bytes;
+            }
+        }
+
+        // TODO(cyrusn): remove arity.
+        public IEnumerable<PackageWithTypeResult> FindPackagesWithType(string name, int arity, CancellationToken cancellationToken)
+        {
+            var database = this.Database;
+            if (database == null)
+            {
+                // Don't have a database to search.  
+                yield break;
+            }
+
+            var query = new MemberQuery(name, isFullSuffix: true, isFullNamespace: false);
+
+            var symbols = new PartialArray<Symbol>(1);
+            if (query.TryFindMembers(database, ref symbols))
+            {
+                var result = new List<PackageWithTypeResult>();
+                foreach (var symbol in symbols)
+                {
+                    var nameParts = new List<string>();
+                    GetFullName(nameParts, symbol.FullName.Parent);
+
+                    if (nameParts.Count > 0)
+                    {
+                        yield return new PackageWithTypeResult(symbol.PackageName.ToString(), name, nameParts);
+                        yield break;
```

Comment:
If we're only returning one, then the method name & return type should be updated.